### ISSUE TITLE
Bug: shapely.strtree.STRtree returns false positives

### ISSUE THREAD

**Issue by NavinF** [Toblerity/Shapely#845]
## Expected behavior and actual behavior.

Expected: "The query method on STRtree returns a list of all geometries in the tree that intersect the provided geometry argument." as documented: https://shapely.readthedocs.io/en/latest/manual.html#str-packed-r-tree

Reality: `STRtree.query(x)` sometimes returns objects that don't intersect `x`. Worse yet, it always returns a non-empty list when certain rectangle-shaped regions of the search space are queried. 

## Steps to reproduce the problem.

1. Create STRtree  from buffered random lines
2. Query STRtree  with another buffered line and assert result[0].intersects(queried_line)
3. Wait for assertion to fail

I'll code a minimum repro later if there is interest. For now, here's my workaround that first queries the R-tree and then double-checks the R-tree query results by brute forcing `intersects` to test if a line segment intersects any obstacles. This solved my problem for now, but kills the log(n) time complexity that I'll need later 

```
def drc_check(self, a :Tuple, b :Tuple, obstacles :STRtree) -> bool:
  """
  Returns True if the (a,b) trace is far enough from obstacles to
  meet design rules
  """
  new_trace_string = LineString([a,b]).buffer(self.trace_width/2 + self.trace_spacing/2)
  intersecting_obstacles = obstacles.query(new_trace_string)
  for o in intersecting_obstacles:
    if o.intersects(new_trace_string):
      return False
    else:
      pass #False positive in STRtree :(
  return True
```

## Operating system

Linux

## Shapely version and provenance

1.7.0 installed from PyPI using pip on Python 3.6

**Comment by sgillies** [comment]
This is a documentation bug. The query method returns objects which have *extents* that intersect with the extent of the given geometry. So, yes, there will be false positives of this kind. We will fix the documentation.

**Comment by NavinF** [comment]
Ah, thanks for confirming. Lemme tack on a performance bug/note to this.

I profiled one run of my program and found that checking for false positives with `intersects` takes 318 seconds whereas all 354836 calls to `strtree.py:50(query)` take 9.2 seconds.  

Oddly enough, some instances of STRtree only return a non-empty list when the query is close to the obstacles. In those cases, double-checking R-tree queries is quick and acceptable. 

Other instances of STRtree consistently return a non-empty list if i query anywhere inside large rectangles covering 1/4 of my search space even though the obstacles (buffered LineString) cover a minuscule area of the search space.  

Seems like chopping up large LineStrings into smaller ones reduces the size of the extents and prevents a diagonal  line from blocking off a huge bounding box that covers the entire geom. But that increases memory usage. 

I wonder if I can tune the R-tree construction to limit the maximum size of leaf nodes (rectangles) and allow a single extent to span many of those small rectangles. 

Or perhaps there is a better way to approach this problem? 

Example of a bad R-tree that degrades to O(n) inside the large white rectangles:

![image](https://user-images.githubusercontent.com/968045/75000375-03051d00-5413-11ea-9b10-ac2433531107.png)

**Comment by sgillies** [comment]
Resolved by #857.
